### PR TITLE
Feature#17 add bootstrap form in views

### LINKED ISSUE #17: Add bootstrap form in form views
# REQUIREMENTS

Implement style and validation to all form views with boostrap-_orm gem.

# DESIGN
Add mockups...

# Acceptance Criteria
List all required acceptance criteria
- [x] Improvement of style
- [x] Validation of inputs
- [x] Error handling
- [x] Improve feedback on form submition / error.

### PR DESCRIPTION
**Pull request by Arkamis**
# Description
What did you implemented/Why did you implemented this:
 - Fix Bootstrap 5 installation
 - Add rails_bootstrap_forms gem
 - Improve error / success feedback after form submission.
 - Improve style in devise forms for new session (login), and reset password views.
 -  Improve form validation before submission.

For more background see ticket: #17 

  ## Screenshots 
If applicable, include screenshots of the results or screenshots that help to see changes

### Login improvement
Screenshot_20210423_150405
### Login error feedback example
Screenshot_20210423_150439
### Reset pasword view with an error example
Screenshot_20210423_150518
### Reset password success feedback
Screenshot_20210423_150536

 # Testing
 Help me how can I test or look at the changes
Follow setup guide, start server and go to the given URL with a browser.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -29,8 +29,11 @@ gem 'jbuilder', '~> 2.7'
 
 # Reduces boot times through caching; required in config/boot.rb
 gem 'bootsnap', '>= 1.4.4', require: false
+gem 'cancancan', '~> 3.2', '>= 3.2.1'
 # Authentication plugin https://github.com/heartcombo/devise
 gem 'devise', '~> 4.7', '>= 4.7.3'
+gem 'rails_admin', '~> 2.1', '>= 2.1.1'
+gem "bootstrap_form", git: "https://github.com/bootstrap-ruby/bootstrap_form.git", branch: "bootstrap-5"
```
Comment (RobertVel666) [inline Gemfile:36]: Forgot to mention what cancancan and rails admin was added for.

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -1,6 +1,14 @@
 class ApplicationController < ActionController::Base
   before_action :configure_permitted_parameters, if: :devise_controller?
 
+  def after_sign_in_path_for(resource)
```
Comment (RobertVel666) [inline app/controllers/application_controller.rb:4]: interesting way of using the paths.

File: app/views/devise/passwords/new.html.erb [hunk 3]
```diff
@@ -1,16 +1,20 @@
...
-
-<%= form_for(resource, as: resource_name, url: password_path(resource_name), html: { method: :post }) do |f| %>
-  <%= render "devise/shared/error_messages", resource: resource %>
-
-  <div class="field">
-    <%= f.label :email %><br />
-    <%= f.email_field :email, autofocus: true, autocomplete: "email" %>
+<div class="container mt-5">
+  <div class="row">
+    <div class="col-xs-1-12">
```
Comment (RobertVel666) [inline app/views/devise/passwords/new.html.erb:3]: Didn't know this sintax for col, nice.

File: db/schema.rb [hunk 4]
```diff
@@ -10,11 +10,18 @@
...
   enable_extension "plpgsql"
 
+  create_table "branches", force: :cascade do |t|
+    t.string "name"
+    t.string "address"
+    t.datetime "created_at", precision: 6, null: false
+    t.datetime "updated_at", precision: 6, null: false
+  end
+
   create_table "users", force: :cascade do |t|
```
Comment (RobertVel666) [inline db/schema.rb:25]: pretty big table mates

**Review by RobertVel666 — APPROVED** [review]
There was some things left out of the description but it seems that you did implement the ones said.

File: app/views/layouts/application.html.erb [hunk 5]
```diff
@@ -7,13 +7,24 @@
...
+      </div>
+    <% end %>
+    <% if alert %>
+      <div class="alert alert-danger alert-dismissible fade show" role="alert">
+        <%= alert %>
+        <button type="button" class="btn-close" data-bs-dismiss="alert" aria-label="Close"></button>
+      </div>
+    <% end %>
+    
     <%= yield %>
```
Comment (RobertVel666) [inline app/views/layouts/application.html.erb:28]: There is no .row and .col

File: app/views/devise/passwords/new.html.erb (hunk 3 above)
Comment (Arkamis) [reply app/views/devise/passwords/new.html.erb:3]: my bad

File: app/views/layouts/application.html.erb (hunk 5 above)
Comment (Arkamis) [reply app/views/layouts/application.html.erb:28]: I dont need those here. I just add alerts for global feedback

**Review by facundotoconas — APPROVED** [review]
good job, looks good!

**Pull request merged**